.github/workflows/test.yml

```diff
@@ -148,3 +148,37 @@ jobs:
     - name: Run the emulation tests
       run: |
         pytest --run-emulation test/test_emulation.py
+
+  test-pyodide:
+    name: Test cibuildwheel building pyodide wheels
+    needs: lint
+    runs-on: ubuntu-24.04
```

Comment:
Do we want to tests other platforms (mostly to ensure some local builds could work) ?
Maybe just a reduced set on other platforms than linux.

Reply:
Building on windows absolutely does not work and will not work until Python gets proper cross builds. Building on macs generally sort of works but we don't test it in any way upstream and cannot test it because mac runners are too expensive. So I think we should just mark it as unsupported. Mac builds would be possible to support if someone decided to fund CI resources for it. But it's likely that nobody will ever care enough.

Or were you thinking that we'd use a docker image or something on these platforms? It's not worth it I don't think. We could check for win32 and make a hard error saying it won't work, and check for Darwin and warn saying we don't actually test this configuration and it is known to be buggy but it sometimes works.

Reply:
It's probably worth erroring out on Windows & adding a warning on macOS then.

I was pleased to see this working out-of-the-box on macOS, maybe I was just lucky then ;-)

Adding a docker image for these platforms only is not worth it. If going the docker way, then I'd say it should be the only option.

Reply:
Well I think macOS is generally working, but it's hard to know since we aren't testing it in our CI.

Reply:
macOS works fine for me locally. I think I fixed or at least reported bugs a year or two ago about macOS, and it seem okay now. I'd prefer to not put a warning in for macOS unless people report issues. Error on Windows (not on the subsystem for Linux, I'd assume) is fine, though.

Reply:
In fact there's already a test running on macOS-13. I will enable the test on macOS-14 as it's what I used to fix all the tests I could.

I will add the error message on Windows.